packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -376,34 +386,44 @@ class NestedScrollViewState extends State<NestedScrollView> {
   ScrollController get outerController => _coordinator!._outerController;
 
   _NestedScrollCoordinator? _coordinator;
+  NestedScrollController? _nestedScrollController;
+
+  bool get _useNestedScrollController => widget.controller is NestedScrollController;
```

Comment:
This looks like a violation of the style guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-interleaving-multiple-concepts-together

Reply:
Now the users need to explicitly pass a boolean parameter to indicate that whether or not the nestedScrollController should be used. The default value is false.